e2e/support/helpers/e2e-visual-tests-helpers.js

```diff
@@ -134,11 +137,36 @@ export function pieSliceWithColor(color) {
 
 export function echartsTooltip() {
   // ECharts may keep two dom instances of the tooltip
-  return cy
-    .findAllByTestId("echarts-tooltip")
-    .filter(":visible")
-    .should("have.length", 1)
-    .eq(0);
+  return cy.findAllByTestId("echarts-tooltip").should($elements => {
+    // Use a custom function to check if the fixed-position tooltip is visible,
+    // as Cypress's ":visible" or "be.visible" fails to identify a fixed-position tooltip as visible.
+    const visibleTooltips = $elements
+      .toArray()
+      .filter(isFixedPositionElementVisible);
+
+    // Assert we have exactly one visible tooltip
+    expect(visibleTooltips).to.have.length(
+      1,
+      "there must be only one visible echarts tooltip",
+    );
+
+    const visibleTooltip = visibleTooltips[0];
+
+    const tooltipContainerStyle = window.getComputedStyle(
+      visibleTooltip.closest(".echarts-tooltip-container"),
+    );
+
+    // (metabase#51904): tooltip container must render above the fold in the Embedding SDK.
+    // ensures that we are using fixed-positioned tooltips.
+    expect(tooltipContainerStyle.position).to.equal("fixed");
+
+    // (metabase#52732): tooltip container must have the correct z-index (4)
+    // this assertion prevents the tooltip from being rendered below charts.
+    expect(Number(tooltipContainerStyle.zIndex)).to.equal(DEFAULT_Z_INDEX);
```

Comment:
We use the current default z-index (4) because we haven't backported Raffi's changes to 52.